activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -38,7 +22,9 @@ def method_missing(*)
       end
       NULL_CONFIG = NullConfig.new # :nodoc:
 
-      attr_accessor :schema_cache
+      def schema_reflection
```

Comment:
I wonder if not renaming this would make the tests a bit less clunky 🤔

Reply:
It would indeed require much less changes to the test suite, but I'd rather bite the bullet now.